nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/StandardProcessorNode.java

```diff
@@ -1321,12 +1325,21 @@ public void run() {
                         this.unscheduled = true;
                     }
                     try {
-                        if (activeThreadMonitorCallback.call()) {
+                        if (scheduleState.isScheduled()) {
+                            schedulingAgent.unschedule(StandardProcessorNode.this, scheduleState);
+                        }
```

Comment:
I am ok with whatever decision but just to clarify; Logic for scheduling/unscheduling was tightly encapsulated in ```StandardProcessScheduler``` where it had access to other things that may be required to perform scheduling/unscheduling operations. A callback could always be modified without introducing any changes to ```ProcessNode```. I am simply afraid that doing the above may lead to dragging more things into ```ProcessNode``` from ```StandardProcessScheduler``` which will eventually lead to package dependency cycles and all the nasty stuff it brings with it.